src/main/java/org/openrewrite/java/spring/security5/UseNewRequestMatchers.java

```diff
@@ -0,0 +1,110 @@
+    private static final MethodMatcher REGEX_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry regexMatchers(..)");
+
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Use the new `requestMatchers` methods";
+    }
+
+    @NonNull
+    @Override
+    public String getDescription() {
+        return "In Spring Security 5.8, the `antMatchers`, `mvcMatchers`, and `regexMatchers` methods were deprecated in favor of new `requestMatchers` methods. Refer to the [Spring Security docs](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#use-new-requestmatchers) for more information.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return Applicability.or(
+                new UsesMethod<>(ANT_MATCHERS),
+                new UsesMethod<>(MVC_MATCHERS),
+                new UsesMethod<>(REGEX_MATCHERS));
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
+                if (ANT_MATCHERS.matches(mi) || MVC_MATCHERS.matches(mi) || REGEX_MATCHERS.matches(mi)) {
+                    mi = maybeChangeMethodInvocation(mi);
+                }
+                return mi;
+            }
+        };
+
+    }
+
+    private J.MethodInvocation maybeChangeMethodInvocation(J.MethodInvocation mi) {
+        JavaType.Method requestMatchersMethod = findRequestMatchersMethodWithMatchingParameterTypes(mi);
+        if (requestMatchersMethod != null) {
+            return mi
+                    .withMethodType(requestMatchersMethod)
+                    .withName(mi.getName().withSimpleName("requestMatchers"));
+        } else {
+            return mi;
+        }
+    }
+
+    @Nullable
+    private JavaType.Method findRequestMatchersMethodWithMatchingParameterTypes(J.MethodInvocation mi) {
+        JavaType.Method methodType = requireNonNull(mi.getMethodType(), "methodType");
+        List<JavaType> parameterTypes = methodType.getParameterTypes();
+        List<JavaType.Method> methods;
+        boolean isOverride = TypeUtils.isOverride(mi.getMethodType());
+        if (isOverride) {
+            methods = requireNonNull(methodType.getDeclaringType().getSupertype(), "superType").getMethods();
+        } else {
+            methods = methodType.getDeclaringType().getMethods();
+        }
+        return methods.stream()
+                .filter(m -> m.getName().equals("requestMatchers"))
+                .filter(m -> m.getParameterTypes().equals(parameterTypes))
+                .findFirst()
+                .orElse(null);
```

Comment:
In terms of method signature it might be preferable to return an Optional indeed; slightly more expressive.

Reply:
Resolved in https://github.com/openrewrite/rewrite-spring/pull/286/commits/28748e9c764026f47ebb3e4a9f8b4494c52dd30c